devito/data/data.py

```diff
@@ -189,8 +200,46 @@ def __str__(self):
     @_check_idx
     def __getitem__(self, glb_idx, comm_type, gather_rank=None):
         loc_idx = self._index_glb_to_loc(glb_idx)
-        is_gather = True if isinstance(gather_rank, int) else False
-        if comm_type is index_by_index or is_gather:
+        is_gather = isinstance(gather_rank, int)
+        if is_gather and comm_type is gather:
+            comm = self._distributor.comm
+            rank = comm.Get_rank()
+
+            sendbuf = np.array(self[:].flatten())
```

Comment:
`flattent` makes a copy no? wouldn something like `self.flat` work here (1D view of ndarray without copy)

Reply:
Yes, indeed. Changed this to `self.flat[:]`.